tests/auto/api.json

```diff
@@ -1266,6 +1266,7 @@
         },
         "DeviceDescriptor": {
             "description": "String",
+            "deviceId": "Uuid",
```

Comment:
This should be optional

Reply:
And why are the tests not failing if this is not optional?

Reply:
Well, it is what the core returns... and current code always sets a deviceId, even though it might be invalid (0000-0000-000000-0000)...

Pro: The client developer can just use it without checking if it's there and match it against its devices (there should never be a device with an null-ID)

Cons: unneeded payload over the network.

Wdyt?

Reply:
The API says it is optional, so either remove the optional and give information what the zero uuid means, or leave the optional and send only the deviceId if there is a device. 
On client logic this would help to understand if this is a edit or a new device. It would be also more clear while reading the API.